op-e2e/actions/interop/proofs_test.go

```diff
@@ -0,0 +1,549 @@
+package interop
+
+import (
+	"fmt"
+	"log/slog"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/trace/super"
+	challengerTypes "github.com/ethereum-optimism/optimism/op-challenger/game/fault/types"
+	"github.com/ethereum-optimism/optimism/op-e2e/actions/helpers"
+	fpHelpers "github.com/ethereum-optimism/optimism/op-e2e/actions/proofs/helpers"
+	"github.com/ethereum-optimism/optimism/op-program/client/claim"
+	"github.com/ethereum-optimism/optimism/op-program/client/interop"
+	"github.com/ethereum-optimism/optimism/op-program/client/interop/types"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInteropFaultProofs(gt *testing.T) {
+	t := helpers.NewDefaultTesting(gt)
+	dsl := NewInteropDSL(t)
+
+	dsl.AddL2Block(dsl.Actors.ChainA)
+	dsl.AddL2Block(dsl.Actors.ChainB)
+
+	dsl.AdvanceSafeHeads(func(opts *AdvanceSafeHeadOpts) {
+		opts.SetChains(dsl.Actors.ChainA)
+	})
+	dsl.AdvanceSafeHeads(func(opts *AdvanceSafeHeadOpts) {
+		opts.SetChains(dsl.Actors.ChainB)
+	})
```

Comment:
nit: I think it can simplified as
```suggestion
	dsl.AdvanceSafeHeads(func(opts *AdvanceSafeHeadOpts) {
		opts.SetChains(dsl.Actors.ChainA, dsl.Actors.ChainB)
	})
```

Reply:
Ah, that was a deliberate change - we want the two chains to submit batches in separate L1 blocks (otherwise we could just use `dsl.AdvanceSafeHeads()` to submit batch data for all chains.  I probably should add a comment and the `AdvanceSafeHeads` method name might need a better name...

Reply:
Yeah it wasn't clear to me it's submitting batches. The name suggests it's updating derivation's view of L1.

Reply:
Went with `SubmitBatchData` and added a comment to explain why we are submitting the chains separately.